src/main/java/org/broadinstitute/hellbender/engine/ShardBoundary.java

```diff
@@ -0,0 +1,90 @@
+package org.broadinstitute.hellbender.engine;
+
+import htsjdk.samtools.util.Locatable;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * Contains information to determine ShardBoundaries
+ */
+public final class ShardBoundary implements Locatable, Serializable {
+    private static final long serialVersionUID = 1L;
+
+    private final SimpleInterval interval;
+    private final SimpleInterval paddedInterval;
+
+    /**
+     * Create a new ShardBoundary from the given intervals
+     * @param interval the interval covered by the shard
+     * @param paddedInterval the interval covered by the shard's padding, must contain the shard interval
+     */
+    public ShardBoundary(final SimpleInterval interval, final SimpleInterval paddedInterval) {
+        Utils.nonNull(interval);
+        Utils.nonNull(paddedInterval);
+        Utils.validateArg(paddedInterval.contains(interval), "interval must be contained within paddedInterval");
+        this.interval = interval;
+        this.paddedInterval = paddedInterval;
+    }
+
+
+    @Override
+    public String getContig() {
+        return paddedInterval.getContig();
```

Comment:
It's a bit inconsistent/troubling that a `Shard's` location according to `Locatable` does not include padding, but `ShardBoundary's` location according to `Locatable` does include padding.